### PR TITLE
Added initial doc for automated tests

### PR DESCRIPTION
**Pull request by taschini**
- As discussed in #203.
- Doctest tests the test used as example and the line-by-line walkthrough.
- Added intersphinx setup for webtest.
- Also mention `(cd doc/build; make doctest)` in the instructions on how to run doctests.

### CODE REVIEW

File: doc/testing.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
+Automating the tests for your App
+=================================
+
+This an introductory guide to writing automated tests for your
+Morepath app.  We assume you've already installed Morepath; if not,
+see the :doc:`installation` section.
+
+In order to carry out the test we'll use WebTest_, which you'll need
```
Comment (faassen) [inline doc/testing.rst]: I'd say people would install webtest in their test dependencies of their project, along with pytest. So I think we should make that the primary explanation with a quick & dirty install webtest & pytest alongside.

File: doc/testing.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Let’s look at a minimal test of a basic app in Morepath:
+
+.. testcode::
+
+  def test_basic():
+      from morepath.tests.fixtures import basic
+      from webtest import TestApp as Client
+
+      c = Client(basic.app())
```
Comment (faassen) [inline doc/testing.rst]: Depending on Morepath fixtures in the test isn't a good idea, I think. Instead it's better to just write a sample app in this text. (besides, the basic fixture is out of date and uses a lowercase app)

File: doc/testing.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+      response = c.get('/foo')
+
+      assert response.body == b'The view for model: foo'
+
+You can save this function into a file and use a test automation tool
+like pytest_ to run it, or you can also invoke it as a regular Python
```
Comment (faassen) [inline doc/testing.rst]: Show how someone would invoke pytest. I think we should just assume pytest is installed and works for this.

**Comment by faassen** [comment]
Thanks! I left a few comments inline.

**Comment by taschini** [comment]
I'll incorporate the feedback.

**Comment by taschini** [comment]
@href, I set a "cookiecutter" label in the sphinx docs that I can use in doc/testing.rst to reference the "canonical place of discussion" of cookiecutter.

I currently set that label in doc/installation.rst.  If you decide at a later stage that the best place for discussing the cookiecutter is in doc/organizing_your_project.rst, doc/quickstart.rst, or an entirely new file, please move that label along.

File: doc/testing.rst (hunk 2 above)
Comment (taschini) [inline doc/testing.rst]: I actually thought we already have a simple app in the docs, that we explain in the detail: the hello world from the quickstart. I'll use that.

File: doc/testing.rst (hunk 3 above)
Comment (taschini) [inline doc/testing.rst]: Done

**Pull request merged**